Social/Facebook/Oauth.php

```diff
@@ -0,0 +1,189 @@
+                $error .= "Error Code: " . $helper->getErrorCode() . "\n";
+                $error .= "Error Reason: " . $helper->getErrorReason() . "\n";
+                $error .= "Error Description: " . $helper->getErrorDescription() . "\n";
+            } else {
+                header('HTTP/1.0 400 Bad Request');
+                $error = 'Bad request';
+            }
+            throw new \Exception($error);
+        }
+
+        // The OAuth 2.0 client handler helps us manage access tokens
+        $oAuth2Client = $fb->getOAuth2Client();
+
+        // Get the access token metadata from /debug_token
+        $tokenMetadata = $oAuth2Client->debugToken($accessToken);
+
+        // Validation (these will throw FacebookSDKException's when they fail)
+        $tokenMetadata->validateExpiration();
+
+        if (!$accessToken->isLongLived()) {
+            // Exchanges a short-lived access token for a long-lived one
+            try {
+                $accessToken = $oAuth2Client->getLongLivedAccessToken($accessToken);
+            } catch (FacebookSDKException $e) {
+                throw new \Exception("<p>Error getting long-lived access token: " . $e->getMessage() . "</p>\n\n");
+            }
+        }
+
+        try {
+            // Returns a `Facebook\FacebookResponse` object
+            $response = $fb->get('/me', $accessToken);
+        } catch (FacebookResponseException $e) {
+            throw new \Exception('Graph returned an error: ' . $e->getMessage());
+        } catch (FacebookSDKException $e) {
+            throw new \Exception('Facebook SDK returned an error: ' . $e->getMessage());
+        }
+
+        $userdata = [];
+
+        $userdata["user_id"] = $response->getDecodedBody()["id"];
+        $userdata["access_token"] = $accessToken->getValue();
+
+        return $userdata;
+    }
+
+    /**
+     * @param $userid
+     * @param $access_token
+     * @param $link
+     * @param $message
+     * @return \Facebook\GraphNodes\GraphNode
+     * @throws \Exception
+     */
+    public function post($userid, $access_token, $link, $message)
+    {
+        $fb = new Facebook([
+            'facebook_app_id' => $this->facebook_app_id,
+            'app_secret' => $this->facebook_app_id_secret,
+            'default_graph_version' => 'v2.2',
+        ]);
+
+        $postData = [
+            'link' => $link,
+            'message' => $message
+        ];
+
+        try {
+            // Returns a `Facebook\FacebookResponse` object
+            $response = $fb->post('/'. $userid .'/feed', $postData, $access_token);
+        } catch (FacebookResponseException $e) {
+            throw new \Exception('Graph returned an error: ' . $e->getMessage());
+        } catch (FacebookSDKException $e) {
+            throw new \Exception('Facebook SDK returned an error: ' . $e->getMessage());
+        }
+
+        $graphNode = $response->getGraphNode();
+
+        return $graphNode;
```

Comment:
The only location this is called from does nothing with the return from this method so is it even needed ?